applications/GeoMechanicsApplication/custom_utilities/transport_equation_utilities.hpp

```diff
@@ -0,0 +1,68 @@
+// KRATOS___
+//     //   ) )
+//    //         ___      ___
+//   //  ____  //___) ) //   ) )
+//  //    / / //       //   / /
+// ((____/ / ((____   ((___/ /  MECHANICS
+//
+//  License:         geo_mechanics_application/license.txt
+//
+//  Main authors:    Ignasi de Pouplana,
+//                   Vahid Galavi
+//                   Gennady Markelov
+//
+
+#pragma once
+
+// Project includes
+
+// Application includes
+#include "geo_mechanics_application_variables.h"
+
+namespace Kratos
+{
+
+class GeoTransportEquationUtilities
+{
+public:
+    using VectorType = Vector;
+    using MatrixType = Matrix;
+
+    //----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
+    template <unsigned int TDim, unsigned int TNumNodes>
+    static inline void CalculatePermeabilityMatrixH(BoundedMatrix<double, TNumNodes, TNumNodes>& rPMatrix,
```

Comment:
I think it might be possible to make the MatrixType the template argument here, such that both BoundedMatrix and Matrix can be handled in a single function. @avdg81 did something similar for the element utilities (see https://github.com/KratosMultiphysics/Kratos/blob/f8087c739d57934b07a1c7114054fba58909e1fe/applications/GeoMechanicsApplication/custom_utilities/element_utilities.hpp#L600-L611

Not sure, but we might be able to have a similar approach here.

Reply:
In addition to the above suggestion, I'd like to return the function's result using the return type (like you did for the overload) rather than through an output parameter.

Furthermore, I would leave out `H` of the function name (i.e. change it to `CalculatePermeabilityMatrix`), since in other places we also don't include the symbols of the matrices (e.g. `CalculateMassMatrix` rather than `CalculateMassMatrixM`).

Reply:
There are still two functions; however, only one body. There is a question on functionality. Anne suggested to test later efficiency of BoundedMatrix vs Matrix. I have found a link which discussed this topic but the info can be very outdated. In a few words, there was no difference between BoundedMatrix and Matrix for a large matrices. 

The function name has lost H and it returns a matrix now.